### ISSUE TITLE
Form submission

### ISSUE THREAD

**Issue by Niekto350** [prabhuignoto/smart-tagz#6]
Hello, I'm trying to use your plugin (great work by the way :) ) but I have a problem integrating it into my form.

When I add a new tag or delete one of the default tags, the form gets submitted which redirects me to a new page. Also the server does not receive the tags from the form request that get's submitted.

I've googled and looked at your documentation but didn't find any solution. Sorry if it's a stupid question, I'm still a beginner in Vue.

**Comment by Evrim267** [comment]
Hi, i have the same issue. The reaseon i guess you are using plugin inside <form></form> element. If you remove probably it will gonna solve.

**Comment by Niekto350** [comment]
Yes, it's inside a form element. I've also figured out I could probably move it outside of a form and handle it in separate ajax request but I would prefer to keep it inside as It's a part of the form. Regarding the form request I had and idea that I could create another component as a wrapper for smart-tagz and bind the value to some hidden input / hidden multi select (so it would be an array of tags, not a string) but that feels like a hacky solution.

**Comment by prabhuignoto** [comment]
@Niekto350 thanks for reporting the issue.  can you point me to the repo where this issue can be reproduced.

**Comment by Niekto350** [comment]
I don't have one, only a bigger local project where I'm trying to use your plugin as a part of a bigger system. But I think I could try to create one for you.

BTW: What are the chances that I check the issue the same minute as you post your comment? :)

**Comment by Niekto350** [comment]
Okay, I've created a dummy repo for you available at https://github.com/Niekto350/smart-tagz-demo

Just clone it, `npm install` and run `npx mix` to compile it. After that open index.html from to `public` directory. If you click on a (x) button to delete the tag, it get's submitted. If you add a new tag, it also submits the form.

The form should only be submitted by hitting the submit button (or enter at some other form field of course).

**Comment by grantjanecek** [comment]
@Niekto350 Here is a hack that I have bee using

`<smart-tagz
          @keypress.enter.prevent
          @click.prevent
          inputPlaceholder="Create Tags ..."
          :allowPaste="{delimiter: ','}"
          :allowDuplicates="false"
          :maxTags="20"
          :on-changed="setTags"
      />`

**Comment by Niekto350** [comment]
Thank you, I've tried something similar (`@keyup.enter.prevent.stop`) but it wasn't working, I'm still new to Vue so maybe I've did something wrong.

For now I've rewritten the tagging system plugin into another custom vue component built on top of different pure js library as I needed to get it working and could not wait longer for the fix. Once this get's fixed, I may come back to it and rewrite it once again to the original smart-tagz solution.

**Comment by raimondslindezutzu** [comment]
> @Niekto350 Here is a hack that I have been using
> 
> ```
> <smart-tagz
>   @keypress.enter.prevent
>   @click.prevent
> />
> ```

`@click.prevent`

Works like a charm!

`            <smart-tagz v-if="loaded" autosuggest inputPlaceholder="Select Tags..." :sources="formData.tags"
                        :maxTags="5"
                        :on-changed="updateTags"
                        @click.prevent
            />`

`    updateTags(tags) {
      this.tags = tags
    },`

**Comment by raimondslindezutzu** [comment]
I ended up with a custom component and binding to a v-model

`            <smart-tags-component v-if="loaded"
                                  :maxTags="5"
                                  :sources="formData.tags"
                                  v-model="tags"
            />`


`<template>
  <smart-tagz autosuggest
              inputPlaceholder="Select Tags..."
              :sources="sources"
              :maxTags="maxTags ? maxTags : 5"
              :on-changed="updateTags"
              @click.prevent
  />
</template>

<script>
import {SmartTagz} from "smart-tagz";
import "smart-tagz/dist/smart-tagz.css";

export default {
  name: "SmartTagsComponent",
  props: ['sources','maxTags'],
  emits: ['update'],
  methods: {
    updateTags(tags) {
      this.$emit('update:modelValue', tags)
    }
  },
  components: {
    SmartTagz
  },
}
</script>

<style scoped>

</style>`
